index.js

```diff
@@ -111,7 +111,9 @@ function deepmergeConstructor (options) {
     for (i = 0, il = sourceKeys.length; i < il; ++i) {
       isNotPrototypeKey(key = sourceKeys[i]) &&
       (
-        key in target && (targetKeys.indexOf(key) !== -1 && (result[key] = _deepmerge(target[key], source[key])), true) || // eslint-disable-line no-mixed-operators
+        key in target && (targetKeys.indexOf(key) !== -1 && // eslint-disable-line no-mixed-operators
+          ((isMergeableObject(source[key]) && cloneProtoObject && Object.getPrototypeOf(source[key]) !== JSON_PROTO && (result[key] = cloneProtoObject(source[key]))) ||
+          (result[key] = _deepmerge(target[key], source[key]))), true) || // eslint-disable-line no-mixed-operators
         (result[key] = clone(source[key]))
```

Comment:
This is illegible. Could you refactor it into legible `if` or `switch` statements?

Reply:
Take a look now. Hopefully I got it right thanks to the tests. But it was truly a confusing piece of code.

Reply:
Yeah, I know you were just following what was laid down previously. Thank you for taking on the clean up.